view/frontend/web/js/model/spi/callbacks/on-click-payment-order-callback.js

```diff
@@ -0,0 +1,29 @@
+define([], function () {
+    'use strict';
+    return async function (pageSource) {
+        console.log(pageSource);
+        if (pageSource !== 'product') {
+            return;
+        }
+
+        const productAddToCartForm = document.getElementById('product_addtocart_form');
+
+        const productAddToCartUrl = productAddToCartForm.getAttribute('action');
+        const addToCartFormData = new FormData(productAddToCartForm);
+        addToCartFormData.append('source', 'expresspay');
+
+        if(!(jQuery("#product_addtocart_form").validation('isValid')))
```

Comment:
I'm worried this would break if someone changes the ID of this element. Is there a better way of checking the validation of the product using a model or JS class which handles this on the actual product page?

Reply:
The element name 'shouldnt' change as its the default name within magento2. fetching this without using a query selector is something that I'm not sure about within M2. @JosephLeedy might have a better idea on what we can do here as an alternative.

Reply:
This seems to be the way core implements this validation: https://github.com/magento/magento2/blob/2.4/app/code/Magento/Catalog/view/frontend/web/js/validate-product.js

Which utilizes https://github.com/magento/magento2/blob/9c48c556c7c7f07ebee95ce12af541e356abc200/app/code/Magento/Catalog/view/frontend/web/js/catalog-add-to-cart.js#L16, which doesn't do any validation as far as I can see.

But there are mixins to this which also do additional validation such as https://github.com/magento/magento2/blob/9c48c556c7c7f07ebee95ce12af541e356abc200/app/code/Magento/ConfigurableProduct/view/frontend/web/js/catalog-add-to-cart-mixin.js#L14, and other extensions could do the same.

Reply:
Could we call the same `catalogAddToCart` method that `mage.productValidate` calls from my links above? Seems like it'll do what you're wanting to achieve. You could just inject a physical element into the form prior to submitting and it _should_ get picked up.

Reply:
We're planning to revisit this implementation after we satisfy the deadline for this work. I told Josh to use roughly the same logic that I used in the Magento 1 extension so that we could get this done quickly.